download.js

```diff
@@ -35,75 +34,101 @@ const chooseAsset = (assets) => {
+    );
+  }
+
   const requiredCommands = ["wget", "tar"];
   const missingCommands = [];
 
-  (await Promise.all(requiredCommands.map(commandExists))).filter(
-    (exists, index) => {
-      if (!exists) missingCommands.push(requiredCommands[index]);
-    }
+  const existsArr = await Promise.all(
+    requiredCommands.map(cmd => commandExists(cmd))
   );
+  existsArr.forEach((exists, i) => {
+    if (!exists) missingCommands.push(requiredCommands[i]);
+  });
 
   if (missingCommands.length > 0) {
     throw new Error(
       `Missing required commands: ${missingCommands.join(", ")}.`
     );
   }
-
-  if (await commandExists("fd")) {
-    throw new Error(
-      `Command 'fd' already exists in your $PATH, so nothing will be installed.`
-    );
-  }
 }
 
 function main() {
-  const req = request(
-    {
-      hostname: "api.github.com",
-      path: "/repos/sharkdp/fd/releases/latest",
-      method: "GET",
-      headers: {
-        "User-Agent":
-          "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.131 Safari/537.36",
-        Accept:
-          "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
-      },
+  const requestOptions = {
+    hostname: "api.github.com",
+    path: "/repos/sharkdp/fd/releases/latest",
+    method: "GET",
+    headers: {
+      "User-Agent":
+        "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.131 Safari/537.36",
+      Accept:
+        "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
     },
-    (res) => {
-      let data = "";
-
-      res.on("data", (_) => (data += _));
-      res.on("end", (_) => {
-        let { assets } = JSON.parse(data);
-        const asset = chooseAsset(assets);
-        downloadAsset(asset);
-      });
-    }
-  );
-  req.on("error", console.log);
+  };
+  const req = request(requestOptions, res => {
+    let data = "";
+
+    res.on("data", _ => (data += _));
+    res.on("end", _ => {
+      if (res.statusCode !== 200) {
+        throw new Error(`Request failed with status code ${res.statusCode}`);
+      }
+      const obj = JSON.parse(data);
+      if (!obj || !Array.isArray(obj.assets)) {
+        throw new Error("no assets found in response");
+      }
+      const asset = chooseAsset(obj.assets);
+      downloadAsset(asset);
+    });
+  });
+  req.on("error", e => {
+    console.error(e);
+    console.error(
+      "Error while downloading asset list. Please check your internet connection."
+    );
+  });
   req.end();
 }
 
-checkRequirements().then(main);
+(async function () {
+  try {
+    await checkRequirements();
+    main();
+  } catch (e) {
+    process.exitCode = 1;
+    if (e instanceof Error) console.error(e.message);
+    else console.error(e);
+  }
+})();
```

Comment:
I made the `main` function to avoid this, can you rename the function to `installFd` or something like that and then change this one to `main`?

```suggestion
async function main() {
  await checkRequirements();
  installFd();
}

main()
  .catch(e => {
    process.exitCode = 1;
    if (e instanceof Error) console.error(e.message);
    else console.error(e);
  })
```

Reply:
This is added in [d2a01f3](https://github.com/pablopunk/fd-find/pull/4/commits/d2a01f3ac827c35fde204b6f09024546349c00aa)